### PR TITLE
update editorconfig, enforce via CI

### PR DESCRIPTION
**Pull request by prusnak**
This PR enables more options in the editorconfig file:
- end of line = LF (unix)
- ~insert final newline = true~
- trim trailing whitespace = true
- charset = utf-8

I have not reformatted any existing files. The changes will be slowly applied by people editing the files with editors that support editorconfig.
Also there is no check (via editorconfig-checker) in the CI right now.

### CODE REVIEW

**Comment by prusnak** [comment]
The CI failure is intermittent and not related to the PR.

**Comment by wozeparrot** [comment]
Should probably add a check in CI.

**Comment by prusnak** [comment]
> Should probably add a check in CI.

I didn't want to add a check in the CI, because then I would need to fix all the files now creating a large changeset containing only whitespace changes touching lot of files.

If this is preferred, let me know and I'll update the PR, I just thought this approach was better:

>  The changes will be slowly applied by people editing the files with editors that support editorconfig.

**Comment by wozeparrot** [comment]
I think it's better if we just apply all the fixes now, and add the check in CI, so that this stuff is enforced. Basically only gonna merge style *enforcement* prs.

**Comment by geohot** [comment]
I don't want to enforce 'insert final newline'

**Comment by prusnak** [comment]
I pushed the requested changes into the repo.

The CI didn't run, because I assume it doesn't run when a contributor touches the github workflow file, so I couldn't have this tested in the github CI.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -27,6 +27,9 @@ jobs:
       run: pip install -e '.[linting,testing]' --extra-index-url https://download.pytorch.org/whl/cpu
     - name: Repo line count
       run: python3 sz.py
+    - name: EditorConfig Checker
+      uses: editorconfig-checker/action-editorconfig-checker@main
+      run: editorconfig-checker
```
Comment (wozeparrot) [inline .github/workflows/test.yml:32]: Nope, the CI didn't run cuz the workflow file is broken, can't put the uses and the run into the same step.

**Comment by prusnak** [comment]
> I don't want to enforce 'insert final newline'

Globally or for some specific files?

**Comment by prusnak** [comment]
Updated the PR to not enforce newlines at end of files

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/runtime/ops_metal.py              1               1
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

File: .editorconfig [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+
+# Unix-style newlines with a newline ending every file, utf-8 charset
+[*]
+end_of_line = lf
+trim_trailing_whitespace = true
+charset = utf-8
+
 [*.py]
 indent_style = space
-indent_size = 2
```
Comment (wozeparrot) [inline .editorconfig:4]: why was this removed?

Comment (prusnak) [reply .editorconfig:4]: 1) Because it is now enforced by pylint.

2) Editorconfig does not handle this properly as there are cases when this strict rule needs to be broken: when function argument list is split into two lines, the second line can start on both even and odd column, depending on the function name.
